osu.Framework.Font/Graphics/Shaders/RepeatMovingBackgroundShader.cs

```diff
@@ -2,52 +2,64 @@
 // See the LICENCE file in the repository root for full licence text.
 
 using System;
+using System.Runtime.InteropServices;
 using osu.Framework.Graphics.Textures;
+using osu.Framework.Graphics.Shaders.Types;
 using osuTK;
-using osuTK.Graphics.ES30;
 
 namespace osu.Framework.Graphics.Shaders;
 
-public class RepeatMovingBackgroundShader : InternalShader, IHasCurrentTime, IHasTextureSize
+[StructLayout(LayoutKind.Sequential, Pack = 1)]
+public record struct RepeatMovingBackgroundParameters
+{
+    public UniformVector2 TexSize;
+    public UniformVector2 RepeatSampleCoord;
+    public UniformVector2 RepeatSampleSize;
+    public UniformVector2 DisplaySize;
+    public UniformVector2 DisplayBorder;
+    public UniformVector2 Speed;
+    public UniformFloat Time;
+    public UniformFloat Mix;
+}
+
+public class RepeatMovingBackgroundShader : CustomizedShader<RepeatMovingBackgroundParameters>, IHasCurrentTime, IHasTextureSize
 {
     public override string ShaderName => "RepeatMovingBackground";
 
     public Texture? Texture { get; set; }
 
+    public Vector2 TextureSize { get; set; }
+
     public Vector2 TextureDisplaySize { get; set; } = new(10);
 
     public Vector2 TextureDisplayBorder { get; set; }
 
     public Vector2 Speed { get; set; }
 
+    public float CurrentTime { get; set; }
+
     public float Mix { get; set; } = 1f;
 
     public override void ApplyValue()
     {
         if (Texture == null)
             return;
 
+        // where this 1 is from? I have no clue.
         Texture.Bind(1);
```

Comment:
Follow the change in the https://github.com/ppy/osu-framework/commit/9ec6711a909cf36ad7c82f12405a20071bf975a5#diff-ec8681ce3076c551c70eeebdc49697c9efb11b7d6f1e10bf0240235dbff7f7d3R122-R138